### PR TITLE
Fix error message when adding new credit card

### LINKED ISSUE #1149: Validation error upon adding payment method even when WCPay not selected
When adding a payment method from the "My Account" front-end page, WooCommerce Payments appears to initiate payment method creation (triggering a validation error if invalid) even if it is not the selected payment method.

Upon clicking "Add payment method", an error shows up if valid card info is not provided in the WooCommerce Payments fields, even if they are hidden:

validation-error-wrong-method

The error does not affect form submission but would still be misleading and confusing to the customer.

**How to simulate:**
- Have WC Payments installed and configured
- Have some other Credit Card Payment method installed and configured
- Log in as a customer
- Go to My account > Payment Methods
- Click on Add Payment Method
- Fill the other Credit Card Payment method
- Try to submit
- The error "Your card number is incomplete" will show up even though you did not select the WC Payments payment method

### PR DESCRIPTION
**Pull request by leonardola**
When a customer adds a new credit card inside My account > Payment Methods using other gateways than WC Payments the error "Your credit card number is invalid" would pop up but the credit card would be added anyways as the error was not related to the payment gateway selected.

Fixes #1149 

#### Changes proposed in this Pull Request
When the customer clicked on Add payment method, WC Payments validated if the information was correctly filled on WC Payments fields, even when the customer selected other Payment Gateway. I've added a conditional to make sure we only validate WC Payments fields if the customer selected a WC Payments Gateway.

Also added tests to guarantee this problem does not happen again in the future.

#### Testing instructions

- Have WC Payments installed and configured
- Have some other Credit Card Payment method installed and configured
- Log in as a customer
- Go to My account > Payment Methods
- Click on Add Payment Method
- Fill the **other gateway** Credit Card details
- Try to submit
- The error "Your card number is incomplete" should not show up

-------------------

- [x] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)

**Post merge**

- [ ] Added testing instructions to the [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) (or does not apply)

### CODE REVIEW

**Comment by leonardola** [comment]
There are some tests failing but they don't seem related to my changes and they fail even when I revert all my code changes. There's some discussion going on about e2e tests failing and a PR trying to fix it [here](https://github.com/Automattic/woocommerce-payments/pull/2992) I've merged develop now but the tests still not pass

File: tests/e2e/specs/shopper/shopper-myaccount-payment-methods-add-fail.spec.js [hunk 1]
```diff
@@ -55,4 +55,31 @@ describe( 'Payment Methods', () => {
 			} );
 		}
 	);
+
+	it( 'to no now show error when adding card on another payment gateway', async () => {
```
Comment (ricardo) [inline tests/e2e/specs/shopper/shopper-myaccount-payment-methods-add-fail.spec.js:59]: How about renaming this test case to `should not show error when adding payment method on another gateway` ?

**Review by ricardo — CHANGES_REQUESTED** [review]
Tests well and looks good.

We should probably add the same check for UPE [here](https://github.com/Automattic/woocommerce-payments/blob/cb8cd200b73ec25cc12237e011e132445544928a/client/checkout/classic/upe.js#L645) as well. Even though UPE doesn't show an error, it also proceeds to validate the form.

**Review by ricardo — APPROVED** [review]
LGTM 👍 

Feel free to merge after fixing the conflict in the changelog.

**Pull request merged**